socketcan_interface/include/socketcan_interface/socketcan.h

```diff
@@ -200,7 +203,7 @@ class SocketCANInterface : public AsioDriver<boost::asio::posix::stream_descript
                 input_.id = frame_.can_id & CAN_EFF_MASK;
                 input_.is_error = 1;
 
-                LOG("error: " << input_.id);
+                ROSCANOPEN_ERROR("socketcan_interface", "internal error: " << input_.id);
```

Comment:
I'd suggest to print the output of `translateError()`, since it is more meaningful than interpreting the id as CAN frame:

```suggestion
                std::string error_description;
                translateError(input_.id, error_description);
                ROSCANOPEN_ERROR("socketcan_interface", "internal error: " << error_description);
```

Or maybe move this into CanLayer, where the output is duplicated...

Reply:
On the other hand, the "controller problems" are output as device state afterwards in can_layer.h, so maybe both outputs in CanLayer and socketcan.h could be dropped...
```
[ INFO] [1562593491.507573769]: Current state: 2 device error: system:0 internal_error: 4 (controller problems;)
[ INFO] [1562593491.507604879]: Current state: 1 device error: system:0 internal_error: 4 (controller problems;)
```